### PR TITLE
Propagate usage of controller-runtime client

### PR DESCRIPTION
**Pull request by ialidzhikov**
**What this PR does / why we need it**:
Propagate usage of controller-runtime client.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```improvement operator
NONE
```

### CODE REVIEW

**Review by timuthy — COMMENTED** [review]
:thumbsup:
/lgtm

Let's merge #1106 first and afterwards adjust the `list` function calls.

**Comment by ialidzhikov** [comment]
> 👍
> /lgtm
> 
> Let's merge #1106 first and afterwards adjust the `list` function calls.

Yep, sure!

File: pkg/operation/botanist/botanist.go [hunk 1]
```diff
@@ -109,7 +110,11 @@ func (b *Botanist) RegisterAsSeed(protected, visible *bool, minimumVolumeSize *s
 		Data: secretData,
 	}
 
-	if _, err := b.K8sGardenClient.CreateSecretObject(secret, true); err != nil {
+	err = b.K8sGardenClient.Client().Create(context.TODO(), secret)
```
Comment (rfranzke) [inline pkg/operation/botanist/botanist.go]: Can you use `CreateOrUpdate`?

File: pkg/operation/botanist/controlplane.go [hunk 2]
```diff
@@ -59,7 +60,12 @@ func (b *Botanist) DeployNamespace() error {
 				gardencorev1alpha1.ShootProvider:  string(b.Shoot.CloudProvider),
 			},
 		},
-	}, true)
+	}
+
+	err := b.K8sSeedClient.Client().Create(ctx, namespace)
```
Comment (rfranzke) [inline pkg/operation/botanist/controlplane.go]: `CreateOrUpdate`

File: pkg/operation/botanist/controlplane.go [hunk 3]
```diff
@@ -82,33 +88,44 @@ func getShootAnnotations(annotations map[string]string, uid types.UID) map[strin
...
 		ObjectMeta: metav1.ObjectMeta{
 			Name: common.GenerateBackupNamespaceName(b.BackupInfrastructure.Name),
 			Labels: map[string]string{
 				common.GardenRole: common.GardenRoleBackup,
 			},
 		},
-	}, true)
+	}
+
+	err := b.K8sSeedClient.Client().Create(ctx, namespace)
```
Comment (rfranzke) [inline pkg/operation/botanist/controlplane.go]: `CreateOrUpdate`

File: pkg/operation/botanist/controlplane.go [hunk 4]
```diff
@@ -124,17 +141,24 @@ func (b *Botanist) DeleteDeprecatedCloudMetadataServiceNetworkPolicy(ctx context
...
-func (b *Botanist) DeleteKubeAPIServer() error {
-	err := b.K8sSeedClient.DeleteDeployment(b.Shoot.SeedNamespace, common.KubeAPIServerDeploymentName)
+func (b *Botanist) DeleteKubeAPIServer(ctx context.Context) error {
+	deploy := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      common.KubeAPIServerDeploymentName,
+			Namespace: b.Shoot.SeedNamespace,
+		},
+	}
+	err := b.K8sSeedClient.Client().Delete(ctx, deploy, kubernetes.DefaultDeleteOptionFuncs...)
```
Comment (rfranzke) [inline pkg/operation/botanist/controlplane.go]: After #1106 is merged you can use `client.IgnoreNotFound`

File: pkg/controllermanager/controller/project/project_control_delete.go [hunk 5]
```diff
@@ -70,7 +73,7 @@ func (c *defaultControl) deleteNamespace(project *gardenv1beta1.Project, namespa
 		return true, nil
 	}
 
-	if err := c.k8sGardenClient.DeleteNamespace(namespaceName); err != nil && !apierrors.IsNotFound(err) {
+	if err := c.k8sGardenClient.Client().Delete(context.TODO(), namespace, kubernetes.DefaultDeleteOptionFuncs...); err != nil && !apierrors.IsNotFound(err) {
```
Comment (rfranzke) [inline pkg/controllermanager/controller/project/project_control_delete.go]: After #1106 is merged you can use `client.IgnoreNotFound`

File: pkg/controllermanager/controller/seed/seed_control.go [hunk 6]
```diff
@@ -192,7 +193,7 @@ func (c *defaultControl) ReconcileSeed(obj *gardenv1beta1.Seed, key string) erro
 				secretFinalizers := sets.NewString(secret.Finalizers...)
 				secretFinalizers.Delete(gardenv1beta1.ExternalGardenerName)
 				secret.Finalizers = secretFinalizers.UnsortedList()
-				if _, err := c.k8sGardenClient.UpdateSecretObject(secret); err != nil && !apierrors.IsNotFound(err) {
+				if err := c.k8sGardenClient.Client().Update(context.TODO(), secret); err != nil && !apierrors.IsNotFound(err) {
```
Comment (rfranzke) [inline pkg/controllermanager/controller/seed/seed_control.go]: After #1106 is merged you can use `client.IgnoreNotFound`

File: pkg/operation/botanist/botanist.go [hunk 7]
```diff
@@ -163,7 +168,13 @@ func (b *Botanist) UnregisterAsSeed() error {
...
 		return err
 	}
-	if err := b.K8sGardenClient.DeleteSecret(seed.Spec.SecretRef.Namespace, seed.Spec.SecretRef.Name); err != nil && !apierrors.IsNotFound(err) {
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      seed.Spec.SecretRef.Name,
+			Namespace: seed.Spec.SecretRef.Namespace,
+		},
+	}
+	if err := b.K8sGardenClient.Client().Delete(context.TODO(), secret, kubernetes.DefaultDeleteOptionFuncs...); err != nil && !apierrors.IsNotFound(err) {
```
Comment (rfranzke) [inline pkg/operation/botanist/botanist.go]: After #1106 is merged you can use `client.IgnoreNotFound`

I probably missed a lot of places, can you check yourself? :)

**Comment by ialidzhikov** [comment]
Thank you for the reviews! PTAL :)

**Review by rfranzke — APPROVED** [review]
/lgtm

**Pull request merged**